scripts/common/repository.js

```diff
@@ -112,46 +110,107 @@ class Repository {
-      bump({
-        preset: 'angular',
-        lernaPackage: this.name,
-        path: this.location,
-      }, (err, recommendation) => {
-        if (err) {
-          reject(err);
+  bump(prerelease = false, preid = null) {
+    return this.constructor.fetchTags()
+      .then((...args) => this.createSmokeTag(args))
+      .then(() => new Promise((resolve, reject) => {
+        bump({
+          preset: 'angular',
+          lernaPackage: this.name,
+          path: this.location,
+        }, (err, recommendation) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(recommendation);
+          }
+        });
+      }))
+      .then((recommendation) => {
+        const regExpSemverPreRelease = /(\d+)\.(\d+)\.(\d+)-(\w+)\.(\d+)/g;
+        let newVersion;
+
+        // Check if the last release was a pre-release
+        // If it's already a pre-release, we should sometimes but only the pre-release digit
+        // Because we don't want:
+        // - packageA@2.0.1-alpha.0
+        // - Patch on packageA
+        // - packageA@2.0.2-alpha.0
+        // But:
+        // - packageA@2.0.1-alpha.0
+        // - Patch on packageA
+        // - packageA@2.0.1-alpha.1
+        if (prerelease && regExpSemverPreRelease.test(this.version)) {
+          const matchs = this.version.match(regExpSemverPreRelease);
+          // 0: Fullmatch
+          // 1: Major digit
+          // 2: Minor digit
+          const minorDigit = parseInt(matchs[2], 10);
+          // 3: Patch Digit
+          // 4: Pre-release name
+          // 5: Digit of pre-release
+
+          if (recommendation.releaseType === 'patch') {
+          // Match example: 2.1.1-alpha.0 -> 2.1.1-alpha.1
+          // Match example: 2.1.0-alpha.0 -> 2.1.0-alpha.1
+            newVersion = semver.inc(this.version, 'prerelease', preid);
+          } else if (recommendation.releaseType === 'minor') {
+          // Min Minor
+            if (minorDigit > 0) {
+            // Match example: 2.1.0-alpha.0 -> 2.1.0-alpha.1
+              newVersion = semver.inc(this.version, 'prerelease', preid);
+            } else {
+            // Match example: 2.1.1-alpha.0 -> 2.2.0-alpha.0
+            // Match example: 2.0.1-alpha.0 -> 2.1.0-alpha.0
+              newVersion = semver.inc(this.version, 'preminor', preid);
+            }
+          } else {
+          // Min major
+          // Match example: 3.0.0-alpha.0 -> 3.0.0-alpha.1
+          // Match example: 2.1.1-alpha.0 -> 3.0.0-alpha.0
+          // Match example: 2.1.0-alpha.0 -> 3.0.0-alpha.0
+            newVersion = semver.inc(this.version, 'prerelease', preid);
+          }
         } else {
-          resolve(recommendation);
+          newVersion = semver.inc(this.version, prerelease ? `pre${recommendation.releaseType}` : recommendation.releaseType, preid);
         }
+
+        return this.updatePackageJson({ version: newVersion })
+          .then(() => this.deleteSmokeTag())
```

Comment:
```suggestion
          .then(this.deleteSmokeTag)
```

Reply:
No, `this` will be different with your suggestion :(

It's why i use an arrow function here :)

Reply:
Noticed. Thanks.